src/chapters/spring-model-validation/thymeleaf-form-tools.rst

```diff
@@ -0,0 +1,224 @@
+
+Using ``th:errors``
+^^^^^^^^^^^^^^^^^^^
+
+The Thymeleaf attribute ``th:errors`` is used similarly to ``th:field`` to display field-specific error messages. Recall that when we added our validation annotations to each model field, we also :ref:`added a message argument <validation-messages>`. Setting ``th:errors`` to a field will display any validation errors for that field.
+
+For example, let's add a new element to the first form group:
+
+.. sourcecode:: html
+   :lineno-start: 9
+
+   <div class="form-group">
+      <label>Name
+         <input th:field="${event.name}" class="form-control">
+      </label>
+      <p class="error" th:errors="${event.name}"></p>
+   </div>
+
+Setting ``th:errors="${event.name}"`` tells Thymeleaf to insert any error messages related to the ``name`` field of ``event`` into the paragraph element. We add ``class="error"`` to allow us to style this element, for example with red text. A simple rule in our ``styles.css`` file will do the trick:
+
+.. sourcecode:: css
+
+   .error {
+     color: red;
+   }
+
+.. admonition:: Note
+
+   Make sure that ``styles.css`` is included in the ``head`` fragment of ``fragments.html``, or the stylesheet will not load.
+
+Using this attribute on all of the fields gives us our final form template code:
+
+.. sourcecode:: html
+   :lineno-start: 8
+
+   <form method="post">
+      <div class="form-group">
+         <label>Name
+               <input type="text" th:field="${event.name}" class="form-control">
+         </label>
+         <p class="error" th:errors="${event.name}"></p>
+      </div>
+      <div class="form-group">
+         <label>Description
+               <input type="text" th:field="${event.description}" class="form-control">
+         </label>
+         <p class="error" th:errors="${event.description}"></p>
+      </div>
+      <div class="form-group">
+         <label>Contact Email
+               <input type="text" th:field="${event.contactEmail}" class="form-control">
+         </label>
+         <p class="error" th:errors="${event.contactEmail}"></p>
+      </div>
+      <div class="form-group">
+         <input type="submit" value="Create" class="btn btn-success">
+      </div>
+   </form>
+
+Now, when the form is submitted with invalid data, our custom validation error messages will display just below the given inputs.
+
+.. figure:: figures/display-validation-errors.png
+   :alt: Our Create Event form after submission with all fields blank. Red error messages are visible next to the fields that failed validation.
+   :width: 700px
+
+   The result of submitting an empty form
```

Comment:
cc option:
Which HTML attributes will a th:field attribute not influence?
#. id
#. name
#. value
#. field